command/server.go

```diff
@@ -1641,39 +1641,86 @@ func (c *ServerCommand) Run(args []string) int {
 			logWriter := c.logger.StandardWriter(&hclog.StandardLoggerOptions{})
 			pprof.Lookup("goroutine").WriteTo(logWriter, 2)
 
-			if os.Getenv("VAULT_STACKTRACE_WRITE_TO_FILE") != "" {
-				c.logger.Info("Writing stacktrace to file")
-
-				dir := ""
-				path := os.Getenv("VAULT_STACKTRACE_FILE_PATH")
-				if path != "" {
-					if _, err := os.Stat(path); err != nil {
-						c.logger.Error("Checking stacktrace path failed", "error", err)
-						continue
-					}
-					dir = path
-				} else {
-					dir, err = os.MkdirTemp("", "vault-stacktrace")
-					if err != nil {
-						c.logger.Error("Could not create temporary directory for stacktrace", "error", err)
-						continue
-					}
+			if os.Getenv("VAULT_PPROF_SIGUSR2_DUMP") != "" {
+				dir, err := os.MkdirTemp("", "pprof")
+				if err != nil {
+					c.UI.Error(err.Error())
+					continue
+				}
+
+				fGoRoutine, err := os.Create(fmt.Sprintf("%s/goroutines", dir))
+				if err != nil {
+					c.UI.Error(err.Error())
+					continue
+				}
+
+				err = pprof.Lookup("goroutine").WriteTo(fGoRoutine, 0)
+				if err != nil {
+					c.UI.Error(err.Error())
+					continue
+				}
+
+				fHeap, err := os.Create(fmt.Sprintf("%s/heap", dir))
+				if err != nil {
+					c.UI.Error(err.Error())
+					continue
+				}
+
+				err = pprof.Lookup("heap").WriteTo(fHeap, 0)
+				if err != nil {
+					c.UI.Error(err.Error())
+					continue
+				}
+
+				fAllocs, err := os.Create(fmt.Sprintf("%s/allocs", dir))
+				if err != nil {
+					c.UI.Error(err.Error())
+					continue
+				}
+
+				err = pprof.Lookup("allocs").WriteTo(fAllocs, 0)
+				if err != nil {
+					c.UI.Error(err.Error())
+					continue
+				}
+
+				fMutex, err := os.Create(fmt.Sprintf("%s/mutex", dir))
```

Comment:
IIRC the mutex and block pprofs won't do anything unless runtime tunables are set first, maybe we shouldn't bother unless that's done?